qiskit/quantum_info/operators/predicates.py

```diff
@@ -55,9 +67,9 @@ def matrix_equal(mat1, mat2, ignore_phase=False, rtol=RTOL_DEFAULT, atol=ATOL_DE
         rtol = RTOL_DEFAULT
 
     if not isinstance(mat1, np.ndarray):
-        mat1 = np.array(mat1)
+        mat1 = np.asarray(mat1)
     if not isinstance(mat2, np.ndarray):
-        mat2 = np.array(mat2)
+        mat2 = np.asarray(mat2)
```

Comment:
If we're calling `asarray` (which I agree we should be), we can skip the `isinstance` checks.